src/components/CaretakerFilters.tsx

```diff
@@ -23,14 +23,25 @@ const CaretakerFilters: React.FC<CaretakerFiltersProps> = ({
 
   const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
     if (e.key === 'Enter') {
-      onSearch();
+      handleSearch();
+    }
+  };
+
+  const handlePriceInput = (animalType: string, priceType: 'minPrice' | 'maxPrice', value: string) => {
+    const regex = /^\d{0,5}?$/;
+    const parsedValue = parseFloat(value) || undefined;
+    
+    if (regex.test(value) && (parsedValue === undefined || parsedValue >= 0)) {
```

Comment:
Podpowiesz dlaczego warunek parsedValue === undefined jest ok?

Reply:
Bo ktoś może usunąć cenę, wtedy jest cena staje sie undefined, ale na taką sytuację też chcemy pozwalać, czyli np ktoś wpisał jakaś cene ale się rozmyślił i nie chce po tym filtrować wiec zostawia to pole puste